src/m365/spp/commands/model/model-apply.ts

```diff
@@ -0,0 +1,148 @@
+import { Logger } from '../../../../cli/Logger.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { spp } from '../../../../utils/spp.js';
+import { urlUtil } from '../../../../utils/urlUtil.js';
+import { validation } from '../../../../utils/validation.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import { ListInstance } from '../../../spo/commands/list/ListInstance.js';
+import commands from '../../commands.js';
+import { z } from 'zod';
+import { globalOptionsZod } from '../../../../Command.js';
+import { zod } from '../../../../utils/zod.js';
+
+const options = globalOptionsZod
+  .extend({
+    contentCenterUrl: z.string()
+      .refine(url => validation.isValidSharePointUrl(url) === true, url => ({
+        message: `'${url}' is not a valid SharePoint Online site URL.`
+      })),
+    webUrl: zod.alias('u', z.string()
+      .refine(url => validation.isValidSharePointUrl(url) === true, url => ({
+        message: `'${url}' is not a valid SharePoint Online site URL.`
+      }))
+    ),
+    id: zod.alias('i', z.string()
+      .refine(id => validation.isValidGuid(id) === true, id => ({
+        message: `${id} is not a valid GUID for option 'id'.`
+      }))
+      .optional()),
+    title: zod.alias('t', z.string().optional()),
+    listTitle: z.string().optional(),
+    listId: z.string().refine(listId => validation.isValidGuid(listId) === true, listId => ({
+      message: `${listId} is not a valid GUID for option 'listId'.`
+    })).optional(),
+    listUrl: z.string().optional(),
+    viewOption: z.enum(['NewViewAsDefault', 'DoNotChangeDefault', 'TileViewAsDefault']).optional()
+  })
+  .strict();
+
+declare type Options = z.infer<typeof options>;
+
+interface CommandArgs {
+  options: Options;
+}
+
+class SppModelApplyCommand extends SpoCommand {
+  public readonly viewOptions: string[] = ['NewViewAsDefault', 'DoNotChangeDefault', 'TileViewAsDefault'];
+
+  public get name(): string {
+    return commands.MODEL_APPLY;
+  }
+
+  public get description(): string {
+    return 'Applies (or syncs) a trained document understanding model to a document library';
+  }
+
+  public get schema(): z.ZodTypeAny | undefined {
+    return options;
+  }
+
+  public getRefinedSchema(schema: typeof options): z.ZodEffects<any> | undefined {
+    return schema
+      .refine(options => !(options.id && options.title), {
+        message: `Specify either 'id' or 'title', but not both`
+      })
+      .refine(options => [options.listTitle, options.listId, options.listUrl].filter(x => x !== undefined).length === 1, {
+        message: `Specify at least one of the following options: 'listTitle', 'listId' or 'listUrl'`
+      });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        await logger.log(`Applying a model to a document library...`);
+      }
```

Comment:
I'm a fan of adding more verbose logging to our commands. I suggest that we log something for about every request we do.

* Assert the content center is valid
* Getting model info
* Getting list info
* Applying model to list